### PR TITLE
parsers: add blueprint parser

### PR DESCRIPTION
**Pull request by GabMus**
You can find the parser [in this repo](https://gitlab.com/gabmus/tree-sitter-blueprint).

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1224,6 +1225,17 @@ list.gitattributes = {
   experimental = true,
 }
 
+list.blueprint = {
+  install_info = {
+    url = "https://gitlab.com/gabmus/tree-sitter-blueprint",
+    files = { "src/parser.c" },
+    branch = "master",
+    requires_generate_from_grammar = false,
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:1233]: these two lines are not needed

File: scripts/minimal_init.lua [hunk 2]
```diff
@@ -4,6 +4,7 @@ vim.cmd [[runtime! plugin/nvim-treesitter.lua]]
 
 vim.cmd [[au BufRead,BufNewFile *.conf set filetype=hocon]]
 vim.cmd [[au BufRead,BufNewFile *.gleam set filetype=gleam]]
+vim.cmd [[au BufRead,BufNewFile *.blp set filetype=blueprint]]
```
Comment (clason) [inline scripts/minimal_init.lua:7]: This is not needed, the script is obsolete.

**Review by clason — CHANGES_REQUESTED** [review]

**Comment by GabMus** [comment]
@clason I applied those changes

File: lua/nvim-treesitter/parsers.lua [hunk 3]
```diff
@@ -18,6 +18,7 @@ local filetype_to_parsername = {
   pandoc = "markdown",
   rmd = "markdown",
   cs = "c_sharp",
+  blp = "blueprint",
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:21]: That's not how this is meant to be used: It maps _filetypes_ to parser names, not _extensions_.

You should add the mapping of `*.blp` to filetype `blueprint` in your config, then the `blueprint` parser will just work out of the box.


(The existing entries are misleading and partially wrong.)

Comment (GabMus) [reply lua/nvim-treesitter/parsers.lua]: Fixed, thanks for the review

File: queries/blueprint/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+(boolean) @boolean
+
+(using) @include
+
+(template) @keyword
+
+(decorator) @attribute
+
+(property_definition (property_name) @parameter)
```
Comment (theHamsta) [inline queries/blueprint/highlights.scm:18]: I don't know the exact context here. But might `@property` be more appropriate here?

Comment (GabMus) [reply queries/blueprint/highlights.scm]: Sure, that's more appropriate. Just changed it.

**Review by theHamsta — APPROVED** [review]

File: lua/nvim-treesitter/parsers.lua [hunk 5]
```diff
@@ -1224,6 +1224,15 @@ list.gitattributes = {
   experimental = true,
 }
 
+list.blueprint = {
+  install_info = {
+    url = "https://gitlab.com/gabmus/tree-sitter-blueprint",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:1229]: for other Gitlab parsers we had sometimes problems without the `.git` suffix during version upgrade. Let's see whether that's still the case

Comment (GabMus) [reply lua/nvim-treesitter/parsers.lua:1229]: I can add the suffix if you think it helps

**Review by clason — APPROVED** [review]

**Pull request merged**